alphafold2_pytorch/alphafold2.py

```diff
@@ -333,12 +333,15 @@ def forward(self, x, context = None, mask = None, context_mask = None, tie_attn_
         if exists(tie_attn_dim):
             q, k, v = map(lambda t: rearrange(t, '(b r) h n d -> b r h n d', r = tie_attn_dim), (q, k, v))
 
-            # when tying row-attention, one cannot have any masked out tokens
             if exists(mask):
-                assert torch.all(mask), 'you cannot have any padding if you are to tie the row attention across MSAs'
-                mask = None
+                mask = rearrange(mask, '(b r) n -> b r n', r = tie_attn_dim)
+                num_rows = (mask.sum(dim = -1) > 0).sum(dim = -1)
+                num_rows = rearrange(num_rows, 'b -> b () () ()')
```

Comment:
Is this required? I thought it would broadcast automatically - but you might've checked already